File: app/src/main/java/com/example/pam_project/lists/lists/components/SpinnerCategoryAdapter.java
package com.example.pam_project.lists.lists.components;

import android.content.Context;
import android.widget.ArrayAdapter;

import com.example.pam_project.lists.categories.components.CategoryInformation;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class SpinnerCategoryAdapter {
    private final ArrayAdapter<String> categories;
    private final Map<String, Long> categoriesMap;

    public SpinnerCategoryAdapter(Context context, int layout) {
        List<String> categoriesNames = new ArrayList<>();
        categoriesMap = new HashMap<>();
        categories = new ArrayAdapter<>(context, layout, categoriesNames);
    }

    public void update(final List<CategoryInformation> model) {
        for (CategoryInformation categoryInformation : model) {
            categories.add(categoryInformation.getTitle());
            categoriesMap.put(categoryInformation.getTitle(), categoryInformation.getId());
        }
        categories.notifyDataSetChanged();
    }

    public String getCategoryById(final long id) {
        for (String categoryName : categoriesMap.keySet()) {
            Long categoryId = categoriesMap.get(categoryName);
            if (categoryId != null && categoryId.equals(id))
                return categoryName;
        }

        return "";
    }

    public ArrayAdapter<String> getCategories() {
        return categories;
    }

    public Map<String, Long> getCategoriesMap() {
        return categoriesMap;
    }
}

File: app/src/main/java/com/example/pam_project/features/categories/spinner/SpinnerCategoryAdapter.kt
package com.example.pam_project.features.categories.spinner

import android.content.Context
import android.widget.ArrayAdapter
import com.example.pam_project.features.categories.list.CategoryInformation
import java.util.*

class SpinnerCategoryAdapter(context: Context?, layout: Int) {
    val categories: ArrayAdapter<String?>
    private val categoriesMap: MutableMap<String?, Long?>

    fun update(model: List<CategoryInformation?>?) {
        for (categoryInformation in model!!) {
            categories.add(categoryInformation?.title)
            categoriesMap[categoryInformation?.title] = categoryInformation?.id
        }
        categories.notifyDataSetChanged()
    }

    fun getCategoryById(id: Long): String? {
        for (categoryName in categoriesMap.keys) {
            val categoryId = categoriesMap[categoryName]
            if (categoryId != null && categoryId == id) return categoryName
        }
        return ""
    }

    init {
        val categoriesNames: List<String?> = ArrayList()
        categoriesMap = HashMap()
        categories = ArrayAdapter(context!!, layout, categoriesNames)
    }
}